src/service.ts

```diff
@@ -0,0 +1,250 @@
+import { Container, Contracts, Enums as AppEnums, Services, Utils as AppUtils } from "@arkecosystem/core-kernel";
+import { Interfaces } from "@arkecosystem/crypto";
+import axios from "axios";
+import os from "os";
+
+import { IOptions } from "./interface";
+import * as messages from "./messages";
+
+const VALID_EVENTS = [
+    "block.applied",
+    "block.forged",
+    "block.reverted",
+    "delegate.registered",
+    "delegate.resigned",
+    "forger.failed",
+    "forger.missing",
+    "forger.started",
+    "peer.added",
+    "peer.removed",
+    "transaction.applied",
+    "transaction.expired",
+    "transaction.forged",
+    "transaction.reverted",
+    "wallet.vote",
+    "wallet.unvote",
+    "round.created",
+    "activedelegateschanged",
+];
+
+const CUSTOM_EVENTS = ["activedelegateschanged"];
+const CUSTOM_EVENT_MAPPING = {
+    activedelegateschanged: AppEnums.BlockEvent.Forged, // AppEnums.RoundEvent.Created,
+};
+
+let LAST_ACTIVE_DELEGATES_CACHED: string[] = [];
+
+@Container.injectable()
+export default class Service {
+    @Container.inject(Container.Identifiers.EventDispatcherService)
+    private readonly emitter!: Contracts.Kernel.EventDispatcher;
+
+    @Container.inject(Container.Identifiers.LogService)
+    private readonly logger!: Contracts.Kernel.Logger;
+
+    @Container.inject(Container.Identifiers.TriggerService)
+    private readonly triggers!: Services.Triggers.Triggers;
+
+    // @Container.inject(Container.Identifiers.DposState)
+    // @Container.tagged("state", "clone")
+    // private readonly dposState!: Contracts.State.DposState;
+
+    @Container.inject(Container.Identifiers.WalletRepository)
+    // why state, blockchain - why not state, clone?
+    @Container.tagged("state", "blockchain")
```

Comment:
The `blockchain` instance is the permanent instance with persisted data. Using the `clone` instance will result in a transient instance that should only be used for applying state changes that should not be persisted, like processing transactions to see if they can be applied or not.